src/actions/sagemaker/sagemaker_infer.ts

```diff
@@ -0,0 +1,356 @@
+      description: "Role ARN for accessing SageMaker and S3",
+    },
+    {
+      name: "region",
+      label: "Region",
+      required: true,
+      sensitive: false,
+      description: "AWS Region for accessing SageMaker",
+    },
+    {
+      name: "user_email",
+      label: "Looker User Email",
+      required: true,
+      description: `
+        Click the button on the right and select 'Email'.
+        This is required for the action to send status emails
+        when training or inference jobs are complete.
+      `,
+      sensitive: false,
+    },
+    {
+      name: "smtpHost",
+      label: "SMTP Host",
+      required: true,
+      sensitive: false,
+      description: "Host for sending emails.",
+    },
+    {
+      name: "smtpPort",
+      label: "SMTP Port",
+      required: true,
+      sensitive: false,
+      description: "Port for sending emails.",
+    },
+    {
+      name: "smtpFrom",
+      label: "SMTP From",
+      required: true,
+      sensitive: false,
+      description: "From for sending emails.",
+    },
+    {
+      name: "smtpUser",
+      label: "SMTP User",
+      required: true,
+      sensitive: false,
+      description: "User for sending emails.",
+    },
+    {
+      name: "smtpPass",
+      label: "SMTP Pass",
+      required: true,
+      sensitive: false,
+      description: "Pass for sending emails.",
+    },
+  ]
+
+  async execute(request: Hub.ActionRequest) {
+
+    logJson("request", request)
+
+    try {
+      // get string inputs
+      const {
+        modelName,
+        bucket,
+        awsInstanceType,
+      } = request.formParams
+
+      const { roleArn } = request.params
+
+      // validate string inputs
+      if (!modelName) {
+        throw new Error("Need SageMaker model name.")
+      }
+      if (!bucket) {
+        throw new Error("Need Amazon S3 bucket.")
+      }
+      if (!awsInstanceType) {
+        throw new Error("Need Amazon awsInstanceType.")
+      }
+      if (!roleArn) {
+        throw new Error("Need Amazon Role ARN for SageMaker & S3 Access.")
+      }
+
+      const numInstances = this.getNumericFormParam(request, "numInstances", 1, 500)
+      const numStripColumns = this.getNumericFormParam(request, "numStripColumns", 0, 2)
+
+      // upload data to S3
+      const date = Date.now()
+      const jobName = `transform-job-${date}`
+      const inputDataKey = `${jobName}/transform-input`
+      const rawDataKey = `${jobName}/transform-raw`
+
+      // store data in S3 bucket
+      await this.uploadToS3(request, bucket, numStripColumns, inputDataKey, rawDataKey)
+
+      // create transform job
+      const client = this.getSageMakerClientFromRequest(request)
+
+      const s3InputPath = `s3://${bucket}/${inputDataKey}`
+      const s3OutputPath = `s3://${bucket}`
+      winston.debug("s3InputPath", s3InputPath)
+      winston.debug("s3OutputPath", s3OutputPath)
+
+      const transformParams: SageMaker.CreateTransformJobRequest = {
```

Comment:
I suppose if you wanted to cut the file size/function sizes down a bit, and DRY up some code between this and the other sagemaker actions, you could factor out the sagemaker and s3 configuration/client code